File: M2/DeveloppementWeb/loginlogout/src/main/java/fr/miage/firstapplijee/metier/Personne.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package fr.miage.firstapplijee.metier;

import java.io.Serializable;

/**
 *
 * @author yvan
 */
public class Personne implements Serializable {
    private boolean monsieur;
    private String nom, prenom;
    private int age;

    public Personne() {
        this(true, "Personne", "Juste", 0);
    }
    
    public Personne(boolean monsieur, String nom, String prenom, int age) {
        this.monsieur = monsieur;
        this.nom = nom;
        this.prenom = prenom;
        this.age = age;
    }
    public boolean isMonsieur() {
        return monsieur;
    }
    public String getNom() {
        return nom;
    }
    public String getPrenom() {
        return prenom;
    }
    public int getAge() {
        return age;
    } 

    public void setMonsieur(boolean monsieur) {
        this.monsieur = monsieur;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public void setAge(int age) {
        this.age = age;
    }        
}

File: L2/Structure de données/Diaporamas/14_TableDeHachage/DemoTableHachageAvecCollision/personne.cpp
/* 
 * File:   personne.cpp
 * Author: yvan
 * 
 * Created on 3 avril 2014, 18:09
 */

#include "personne.h"

personne::personne(const string& nom, const string& prenom) : d_nom(nom), d_prenom(prenom) {
}

personne::personne(const personne& orig) : d_nom(orig.d_nom), d_prenom(orig.d_prenom) {
}

string personne::getNom() const {
  return d_nom;
}

string personne::getPrenom() const {
  return d_prenom;
}

personne& personne::operator=(const personne& a) {
  if (this != &a) {
    d_nom = a.d_nom;
    d_prenom = a.d_prenom;
  }
  return *this;
}
ostream& operator<<(ostream& os, const personne& c) {
  os << c.getPrenom() << ' ' << c.getPrenom();
  return os;
}

